src/lib/Libtpp/tpp_platform.c

```diff
@@ -710,10 +710,15 @@ tpp_sock_resolve_host(char *host, int *count)
 	hints.ai_socktype = SOCK_STREAM;
 	hints.ai_protocol = IPPROTO_TCP;
 
+	tpp_nslookup_mutex_lock();
```

Comment:
Change this to call tpp_lock(&tpp_nslookup_mutex);

Reply:
Since as per document of `getaddrinfo()`, its thread safe. I would suggest to add comment here to explain why we are adding lock/unlock around `getaddrinfo()` so in future it won't confuse ppl/dev.